Lesson_7/ACoder.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Lesson_7;
+
+public class ACoder : ICoder
+{
+    public string Encode(string? s)
+    {
+        if (s is null)
+        {
+            throw new ArgumentNullException();
+        }
+        char [] c = s.ToCharArray();
+        for (int i = 0; i < c.Length; i++)
+        {
+            //int a = (char)c[i] + 1;
+            c[i] = (char)((char)c[i] + 1);
+        }
+        s = c.ToString();
```

Comment:
А вот это говорит о том, что даже не запускали код. Он Вам всегда будет возвращать одно и то же `"System.Char[]"`
```suggestion
        s = new string(c);
```